src/parsing.jl

```diff
@@ -256,6 +256,81 @@ function Base.parse(::Type{ZonedDateTime}, pqv::PQValue{PQ_SYSTEM_TYPES[:int8]})
     TimeZones.unix2zdt(parse(Int64, pqv))
 end
 
+## intervals
+# iso_8601
+_DEFAULT_TYPE_MAP[:interval] = Dates.CompoundPeriod
+const INTERVAL_REGEX = Ref{Regex}()  # set at __init__
+
+function _interval_regex()
+    function _field_match(period_type, number_match="-?\\d+")
+        name = nameof(period_type)
+        letter = first(String(name))
+        return "(?:(?<$name>$number_match)$letter)?"
+    end
+
+    io = IOBuffer()
+    print(io, "^P")
+    for long_type in (Year, Month, Day)
+        print(io, _field_match(long_type))
+    end
+    print(io, "(?:T")
+    for long_type in (Hour, Minute)
+        print(io, _field_match(long_type))
+    end
+    print(io,
+        _field_match(Second, "(?<whole_seconds>-?\\d+)(?:\\.(?<frac_seconds>\\d{1,9}))?"),
+    )
+    print(io, ")?\$")
+
+    return Regex(String(take!(io)))
+end
+
+function Base.parse(::Type{Dates.CompoundPeriod}, pqv::PQValue{PQ_SYSTEM_TYPES[:interval]})
```

Comment:
This parse call will require a few tests to check different cases listed in the docs.

Reply:
Oh, whoops, yeah I totally forgot those.